### PR TITLE
Add notebook for structured output generation with Instructor

### PR DESCRIPTION
**Pull request by S4mpl3r**
This pull request adds a new notebook to the tutorials showcasing how to generate structured JSON outputs with `groq` and `instructor`. The `instructor` library already has a Groq integration and it makes it very easy to work with pydantic models for structured output generation. This notebook is inspired by OpenAI's structured output generation feature, which itself is seemingly inspired by the `instructor` library.

### CODE REVIEW

File: tutorials/structured-output-instructor/structured_output_instructor.ipynb [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+    "\n",
+    "Consider a typical scenario: we want to produce structured JSON data from an LLM. While tools like Python's `json` module allow us to handle this data, they also come with their own set of challenges, such as validating data types and ensuring consistency across outputs. Manually checking these aspects can be tedious and error-prone. LLMs also tend to forget to include a comma or a closing bracket ('}') somewhere in the produced JSON from time to time, which would invalidate the whole JSON output.\n",
+    "\n",
+    "This is where the Instructor library comes into play. If you've been using the recent feature of [structured outputs](https://openai.com/index/introducing-structured-outputs-in-the-api/) by OpenAI, then you'll feel at home using Instructor. By integrating the Instructor library with Groq models, we can simplify the process of generating structured outputs, making it both easier and more reliable. This guide will walk you through setting up this integration, showcasing how to use structured outputs to generate synthetic data for evaluating LLM-powered applications - a powerful and practical use case for LLMs."
```
Comment (hozen-groq) [inline tutorials/structured-output-instructor/structured_output_instructor.ipynb:12]:
```suggestion
    "This is where the Instructor library comes into play. If you've been using the recent feature of [structured outputs](https://openai.com/index/introducing-structured-outputs-in-the-api/) by OpenAI, then you'll feel at home using Instructor. By integrating the Instructor library with models powered by Groq, we can simplify the process of generating structured outputs, making it both easier and more reliable. This guide will walk you through setting up this integration, showcasing how to use structured outputs to generate synthetic data for evaluating LLM-powered applications - a powerful and practical use case for LLMs."
```

File: tutorials/structured-output-instructor/structured_output_instructor.ipynb [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1. A Very Simple Use Case\n",
+    "\n",
+    "Let's dive right into how you can set up the `instructor` library with Groq models to generate structured JSON outputs. We'll keep it simple and straightforward so you can get up and running quickly.\n",
```
Comment (hozen-groq) [inline tutorials/structured-output-instructor/structured_output_instructor.ipynb:21]:
```suggestion
    "Let's dive right into how you can set up the `instructor` library with models powered by Groq to generate structured JSON outputs. We'll keep it simple and straightforward so you can get up and running quickly.\n",
```

File: tutorials/structured-output-instructor/structured_output_instructor.ipynb [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+    "\n",
+    "### Grabbing Your API Key\n",
+    "\n",
+    "Before doing anything, grab your Groq API key from [Groq console](https://console.groq.com/keys) and put it in an `.env` file alongside this notebook:\n",
```
Comment (hozen-groq) [inline tutorials/structured-output-instructor/structured_output_instructor.ipynb:25]:
```suggestion
    "Before doing anything, grab your Groq API Key from [Groq Console](https://console.groq.com/keys). If you don't already have an account with GroqCloud, you can create one for free. Once you have your Groq API Key, put it in an `.env` file alongside this notebook (you can use the `.env.example` file in this directory and just edit the filename to `.env`):\n",
```

File: tutorials/structured-output-instructor/structured_output_instructor.ipynb [hunk 4]
```diff
@@ -0,0 +1,311 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Structured Output with Groq and Instructor\n",
+    "While large language models (LLMs) are often employed for building chatbots or conversational agents, numerous real-world applications require a different approach - one that goes beyond mere dialogue and involves producing structured, machine-readable outputs.\n",
```
Comment (hozen-groq) [inline tutorials/structured-output-instructor/structured_output_instructor.ipynb:8]:
```suggestion
    "While Large Language Models (LLMs) are often employed for building chatbots or conversational agents, numerous real-world applications require a different approach - one that goes beyond mere dialogue and involves producing structured, machine-readable outputs.\n",
```

File: tutorials/structured-output-instructor/structured_output_instructor.ipynb [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As you can see, the model returned an object of type `ResponseModel` and has correctly created 5 examples for our evaluation dataset. Great!\n",
+    "\n",
+    "## Conclusion\n",
+    "\n",
+    "With these generated examples, you can evaluate how well the weather agent utilizes the `get_weather_info` tool. By simulating various scenarios, you can test the agent's ability to handle different contexts, ensuring that it correctly identifies when and how to call the tool with the appropriate parameters. This approach enhances not only the evaluation process but also helps identify edge cases where the agent might struggle, allowing you to refine its performance before deploying it in a real-world environment.\n",
+    "\n",
+    "This notebook demonstrates just one of the powerful use cases of structured outputs in Large Language Models (LLMs). By combining the `instructor` library with Groq, you can effortlessly create a diverse set of structured examples, making your agentic workflows more robust and reliable.\n",
```
Comment (hozen-groq) [inline tutorials/structured-output-instructor/structured_output_instructor.ipynb:281]:
```suggestion
    "This notebook demonstrates just one of the powerful use cases of structured outputs in LLMs. By combining the `instructor` library with Groq, you can effortlessly create a diverse set of structured examples, making your agentic workflows more robust and reliable.\n",
```

File: tutorials/structured-output-instructor/structured_output_instructor.ipynb [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+    "\n",
+    "### Useful Links\n",
+    "- [Groq Cookbook](https://github.com/groq/groq-api-cookbook)\n",
```
Comment (hozen-groq) [inline tutorials/structured-output-instructor/structured_output_instructor.ipynb:284]:
```suggestion
    "- [Groq API Cookbook](https://github.com/groq/groq-api-cookbook)\n",
```

**Comment by hozen-groq** [comment]
Finally had a chance to review this and I just want to say a huge thank you because this is extremely well done and very valuable. Just a few small suggestions for wording that I'd love if you could review and incorporate. Overall, what a great addition to have in our cookbook! Fantastic job! 🚀 Can't wait to merge this and promote to our community.

**Review by S4mpl3r — COMMENTED** [review]
Suggested changes reviewed and incorporated!

**Pull request merged**